v03_pipeline/lib/reference_data/compare_globals.py

```diff
@@ -16,130 +20,114 @@
-        return False
-
-    config_or_dataset_version = hl.eval(
-        parse_dataset_version(
-            dataset_ht,
-            dataset,
-            dataset_config,
-        ),
-    )
-    return joined_ht_version == config_or_dataset_version
-
-
-def ht_path_matches_config(
-    joined_ht_globals: ReferenceDataGlobals,
-    dataset: str,
-    dataset_config: dict,
-) -> bool:
-    joined_ht_path = joined_ht_globals.paths.get(dataset)
-    if joined_ht_path is None:
-        return False
-
-    config_path = get_ht_path(dataset_config)
-    return joined_ht_path == config_path
-
-
-def ht_enums_match_config(
-    joined_ht_globals: ReferenceDataGlobals,
-    dataset: str,
-    dataset_config: dict,
-) -> bool:
-    joined_ht_enums = joined_ht_globals.enums.get(dataset, {})
-    config_enums = dataset_config.get('enum_select', {})
-    return joined_ht_enums == config_enums
-
-
-def ht_selects_match_config(
-    joined_ht: hl.Table,
-    dataset: str,
-    dataset_config: dict,
-    dataset_ht: hl.Table,
-) -> bool:
-    joined_ht_selects = set(joined_ht[dataset])
-    config_selects = set(get_all_select_fields(dataset_ht, dataset_config).keys())
-    return len(config_selects.symmetric_difference(joined_ht_selects)) == 0
+    selects: dict[str, set[str]]
+
+    @classmethod
+    def from_dataset_configs(
+        cls,
+        rdc: ReferenceDatasetCollection,
+        dataset_type: DatasetType,
+        reference_genome: ReferenceGenome,
+    ):
+        paths, versions, enums, selects = {}, {}, {}, {}
+        for dataset in rdc.datasets(dataset_type):
+            dataset_config = CONFIG[dataset][reference_genome.v02_value]
+            dataset_ht = import_ht_from_config_path(dataset_config, reference_genome)
+
+            paths[dataset] = get_ht_path(dataset_config)
+            versions[dataset] = hl.eval(
+                parse_dataset_version(
+                    dataset_ht,
+                    dataset,
+                    dataset_config,
+                ),
+            )
+            enums[dataset] = dataset_config.get('enum_select', {})
+            selects[dataset] = set(
+                get_all_select_fields(dataset_ht, dataset_config).keys(),
+            )
+        return cls(paths, versions, enums, selects)
+
+    @classmethod
+    def from_ht(
+        cls,
+        ht: hl.Table,
+        rdc: ReferenceDatasetCollection,
+        dataset_type: DatasetType,
+    ):
+        rdc_globals_struct = hl.eval(ht.globals)
+        paths = dict(rdc_globals_struct.paths)
+        versions = dict(rdc_globals_struct.versions)
+        enums = dict(rdc_globals_struct.enums)
+
+        selects = {}
+        for dataset in rdc.datasets(dataset_type):
+            if dataset in ht.row:
+                select = ht[dataset]
+                if isinstance(select, hl.StructExpression):
+                    selects[dataset] = set(select)
+                else:
+                    selects[dataset] = set()
```

Comment:
this was the case in 1 of the test rdc tables, I don't expect it to happen in real life.

I put this code in here instead of creating new test tables or updating the ones used in the current tests, but that might be the wrong call

Reply:
ah, the result of my laziness 🤣 .  how much work would it be to change it?  we can maybe make a backlog ticket and punt on it... we probably should have test data that aligns to our real data.

Reply:
I think not too much work now that I am pretty familiar with the structure of the data and the tests. I'll do it as part of this PR

Reply:
ok I just had to change the test_combined_1.ht table - it has 1 row with the correct # and shape of annotations